reader.go

```diff
@@ -424,7 +424,18 @@ func (r *reader) run(ctx context.Context, offset int64, join *sync.WaitGroup) {
 			switch offset, err = r.read(ctx, offset, conn); err {
 			case nil:
 				errcount = 0
+			case NotLeaderForPartition:
+				if err = conn.Close(); err != nil {
+					r.withLogger(func(log *log.Logger) {
+						log.Printf("failed to close connection for partition %d of %s at offset %d", r.partition, r.topic, offset)
+					})
+				}
 
+				conn, start, err = r.initialize(ctx, offset)
+				if err != nil {
+					r.sendError(ctx, err)
+					break readLoop
+				}
```

Comment:
I don't think we should bubble up this error by calling `sendError`, the reader should abstract this problem away and pretend that everything is fine.

How about just doing this?
- log the error
- close the connection
- `break readLoop`

The code will flow back normally to the initialization phase.